homeassistant/components/tuya/light.py

```diff
@@ -526,6 +528,19 @@ def __init__(
         ):
             self._color_temp = int_type
             color_modes.add(ColorMode.COLOR_TEMP)
+            self._white_color_mode = ColorMode.COLOR_TEMP
+        elif (
+            workmode_type
+            := self.find_dpcode(  # If entity does not have color_temp, check if it has work_mode "white"
+                description.color_mode, dptype=DPType.ENUM, prefer_function=True
+            )
+        ):
+            if WorkMode.WHITE.value in workmode_type.range:
+                color_modes.add(ColorMode.WHITE)
+                self._white_color_mode = ColorMode.WHITE
+
+        if not self._white_color_mode:
+            self._white_color_mode = ColorMode.COLOR_TEMP  # default to COLOR_TEMP
```

Comment:
Why do we set this, but we also have this check in the `color_mode` property?

```
        return (
            self._white_color_mode if self._white_color_mode else ColorMode.COLOR_TEMP
        )  # default to COLOR_TEMP
```

I'd suggest to simplify to initialize `_white_color_mode` to `ColorMode.COLOR_TEMP`